report/onionscanreport.go

```diff
@@ -64,8 +66,20 @@ type OnionScanReport struct {
 	SMTPFingerprint string `json:"smtpFingerprint"`
 	SMTPBanner      string `json:"smtpBanner"`
 
+	ProtocolInfoList []ProtocolInfo `json::"protocolInfoList"`
+
 	NextAction string `json:"lastAction"`
-	TimedOut   bool
+	TimedOut   bool   `json:"timedOut"`
+}
+
+type ProtocolInfo struct {
+	Type string      `json:"type"`
+	Port uint        `json:"port:`
+	Info interface{} `json:"info"`
+}
+
+func (osr *OnionScanReport) AddProtocolInfo(protocolType string, protocolPort uint, protocolInfo interface{}) {
```

Comment:
This hasn't been finalized yet, and commits shouldn't introduce it or rely on it.